### PR TITLE
vimc-4577 Replace travis script with github action

### PR DESCRIPTION
**Pull request by EmmaLRussell**
Replace the travis CI script with a github action to:
- install dependencies
- run dependencies
- run tests with codecov
- linting
- upload codecov result

This is simpler than the travis script because we don't need to do anything special to get docker and docker compose to work.

### CODE REVIEW

**Review by Nicolas-Dolan — COMMENTED** [review]
I followed the development instructions in the readme, installed the dependencies, and then was wondering, now what? Perhaps, when you have time, we could look at what the purpose of this repo is and how it functions. There is no rush on this though.

**Comment by EmmaLRussell** [comment]
Sorry the docs are very scant. This repo just provides a thin Python wrapper round a couple of the api functions of OrderlyWeb. This enables a task queue in python that we have in another repo (Montagu task queue) to easily run reports as tasks. None of which is all that important for this PR, which just translates the old travis script to run on gha instead.

> I followed the development instructions in the readme, installed the dependencies, and then was wondering, now what? Perhaps, when you have time, we could look at what the purpose of this repo is and how it functions. There is no rush on this though.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: test
+on: [push]
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python ${{ matrix.python-version }}
```
Comment (mwoodbri) [inline .github/workflows/test.yml]: This variable is not set (no `strategy` section) so will never be interpolated

**Review by mwoodbri — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (EmmaLRussell) [reply .github/workflows/test.yml]: :facepalm: Copypasta strikes again..

**Review by mwoodbri — APPROVED** [review]

**Pull request merged**